search.go

```diff
@@ -126,7 +126,11 @@ func (o *opSearch) ExitSearchMode(revert bool) {
 
 	if revert {
 		o.history.current = o.source
-		o.buf.Set(o.history.showItem(o.history.current.Value))
+		var redrawValue []rune
+		if o.history.current != nil {
```

Comment:
Looks sane but i failed to reproduce it myself with fq and example/readline-demo. When is `o.source` `nil`? there is no history at all?

Reply:
Yep. Try deleting this line:

https://github.com/ergochat/readline/blob/176a740b5c790874212064260272132ad19e2651/example/readline-demo/readline-demo.go#L86

Reply:
Ok! yeap now it crashed on plain main and fq branch 👍